handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -310,7 +310,8 @@ private void replaceHandler(ChannelHandlerContext ctx, Selection selection) {
         this.selection = selection;
         try {
             sslEngine = selection.context.newEngine(ctx.alloc());
-            ctx.pipeline().replace(this, SslHandler.class.getName(), selection.context.newHandler(sslEngine));
+            SslHandler handler = SslContext.newHandler(sslEngine);
```

Comment:
Question for you: The SslHandler has this `sslCloseFuture()`. Should we cancel() it along with releasing the SslEngine (as added with #5745) to stay consistent with the expected behavior (i.e. the SslHandler is supposed to complete it when it's releasing the engine).